ports/stm32/stm32_hal_msp.c

```diff
@@ -278,27 +337,31 @@ void HAL_I2C_MspInit(I2C_HandleTypeDef *hi2c) {
     if (0) {
     #if defined(OMV_I2C1_ID)
     } else if (hi2c->Instance == I2C1) {
-        __HAL_RCC_I2C1_CLK_ENABLE();
         scl_pin = OMV_I2C1_SCL_PIN;
         sda_pin = OMV_I2C1_SDA_PIN;
+        __HAL_RCC_I2C1_CLK_ENABLE();
+        __HAL_RCC_I2C1_CLK_SLEEP_ENABLE();
     #endif
     #if defined(OMV_I2C2_ID)
     } else if (hi2c->Instance == I2C2) {
-        __HAL_RCC_I2C2_CLK_ENABLE();
         scl_pin = OMV_I2C2_SCL_PIN;
         sda_pin = OMV_I2C2_SDA_PIN;
+        __HAL_RCC_I2C2_CLK_ENABLE();
+        __HAL_RCC_I2C2_CLK_SLEEP_ENABLE();
     #endif
     #if defined(OMV_I2C3_ID)
     } else if (hi2c->Instance == I2C3) {
-        __HAL_RCC_I2C3_CLK_ENABLE();
         scl_pin = OMV_I2C3_SCL_PIN;
         sda_pin = OMV_I2C3_SDA_PIN;
+        __HAL_RCC_I2C3_CLK_ENABLE();
+        __HAL_RCC_I2C3_CLK_SLEEP_ENABLE();
     #endif
     #if defined(OMV_I2C4_ID)
     } else if (hi2c->Instance == I2C4) {
-        __HAL_RCC_I2C4_CLK_ENABLE();
         scl_pin = OMV_I2C4_SCL_PIN;
         sda_pin = OMV_I2C4_SDA_PIN;
+        __HAL_RCC_I2C4_CLK_ENABLE();
+        __HAL_RCC_I2C4_CLK_SLEEP_ENABLE();
```

Comment:
N6 ifdef?

Reply:
I don't see the harm in enabling it unconditionally.